system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -351,39 +352,82 @@ class CLdapConfig : implements ILdapConfig, public CInterface
 
         m_timeout = cfg->getPropInt(".//@ldapTimeoutSecs", LDAPTIMEOUT);
 
+        //------------------------------------------------
+        //Get LDAP Admin account username and password
+        // m_sysuser_commonname
+        // m_sysuser_password
+        //------------------------------------------------
+//@@
+        StringBuffer xml;//@@
+        toXML(cfg, xml);//@@
+        DBGLOG("@@%s",xml.str());//@@
+//@@
+        StringBuffer systemUserSecretKey;
+        cfg->getProp(".//@systemUserSecretKey", systemUserSecretKey);//secrets repo key
+        if (!systemUserSecretKey.isEmpty())
+        {
+            StringBuffer sb;
+            cfg->getProp(".//@systemCommonName", sb);
+            if (!sb.isEmpty())
+                WARNLOG("Please specify either systemCommonName or systemUserSecretKey in configuration");
```

Comment:
Warning is a little confusing saying that you have to specify one of them, but really the issue is that they specified both.  Should probably be something like:

 WARNLOG("Please specify either systemCommonName or systemUserSecretKey but not both in configuration");